tests/test_integration/test_cpp_regression.py

```diff
@@ -215,6 +215,49 @@ def run_vertical_advection_dycore(backend, id_version, domain):
         )
 
 
+@register
+@hyp.given(
+    domain=hyp_st.tuples(
+        *(
+            [hyp_st.integers(min_value=1, max_value=32)] * 2
+            + [hyp_st.integers(min_value=16, max_value=32)]
+        )
+    )
+)
+def run_large_k_interval(backend, id_version, domain):
+
+    validate_field_names = ["out_field"]
+    origins = {"in_field": (0, 0, 0), "out_field": (0, 0, 0)}
+    shapes = {k: tuple(domain[i] + 2 * origins[k][i] for i in range(3)) for k in origins.keys()}
```

Comment:
```suggestion
    shapes = {k: tuple(domain[i] + 2 * v[i] for i in range(3)) for k, v in origins.items()}
```
Alternatively iterating over items saves you accessing `origin[k]` 3 times per value of `k`

Reply:
Same as above